pool/pool.go

```diff
@@ -0,0 +1,89 @@
+package pool
+
+import (
+	"hash"
+	"sync"
+
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"golang.org/x/crypto/sha3"
+)
+
+// HasherPool pools bmt Hashers.
+// It provides the ability for the number of hashers to grow
+// according to demand, but will shrink once the minimum defined
+// hashers are put back into the pool.
+type HasherPool interface {
```

Comment:
I would maybe call this just `Interface`.

Reply:
i'll change the `Pooler`. Not really a huge fan of using the `<package>.Interface` convention tbh